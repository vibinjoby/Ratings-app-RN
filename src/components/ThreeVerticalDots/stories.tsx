import React from 'react'
import { View } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import ThreeVerticalDots from './component'
import { StyleSheet } from 'react-native'

storiesOf('Components', module).add('ThreeVerticalDots', () => (
  <View style={styles.container}>
    <ThreeVerticalDots displayDatas={['Delete Review', 'Edit Response']} />
  </View>
))

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
})
