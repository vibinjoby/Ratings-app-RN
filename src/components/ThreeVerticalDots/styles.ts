import { StyleSheet } from 'react-native'

import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    width: 300,
    height: 300,
    right: 0,
  },
  dropdwnWrapper: {
    position: 'absolute',
    right: 0,
    top: 20,
    backgroundColor: Colors.white,
    borderRadius: 5,
    padding: 15,
    shadowColor: Colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,

    elevation: 5,
  },
  threeDotWrapper: {
    alignSelf: 'flex-end',
  },
  text: {
    ...Typography.RegularFont,
    fontSize: 14,
  },
  spacing: {
    marginBottom: 5,
  },
})
