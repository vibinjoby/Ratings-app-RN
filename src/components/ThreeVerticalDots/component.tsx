import React from 'react'
import { useState } from 'react'
import { Text, FlatList, Image, TouchableOpacity, View } from 'react-native'

import styles from './styles'

export interface ThreeVerticalDotsProps {
  testID?: string
  displayDatas: Array<string>
  onOptionPress?: (arg0: number) => void
}

const ThreeVerticalDots: React.FC<ThreeVerticalDotsProps> = ({
  testID,
  displayDatas,
  onOptionPress,
}: ThreeVerticalDotsProps) => {
  const [isPressed, setIsPressed] = useState(false)

  const togglePress = () => setIsPressed((val) => !val)

  const ThreeDots = () => (
    <TouchableOpacity testID={testID} onPress={togglePress} style={styles.threeDotWrapper}>
      <Image source={require('../../assets/moreIc/moreIc.png')} />
    </TouchableOpacity>
  )

  const onOptionSelected = (index: number) => {
    togglePress()
    onOptionPress && onOptionPress(index)
  }

  const DropDown = () => (
    <View style={styles.dropdwnWrapper}>
      <FlatList
        data={displayDatas}
        keyExtractor={(_, index) => index.toString()}
        renderItem={({ item, index }) => (
          <TouchableOpacity
            testID={`option ${index}`}
            activeOpacity={0.7}
            onPress={() => onOptionSelected(index)}
          >
            <Text style={styles.text}>{item}</Text>
          </TouchableOpacity>
        )}
        ItemSeparatorComponent={() => <View style={styles.spacing} />}
      />
    </View>
  )

  return (
    <View style={styles.container}>
      {isPressed && <DropDown />}
      <ThreeDots />
    </View>
  )
}

export default ThreeVerticalDots
