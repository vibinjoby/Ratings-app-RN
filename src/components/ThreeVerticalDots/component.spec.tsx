import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import ThreeVerticalDots from './component'

describe('ThreeVerticalDots', () => {
  it('should render', () => {
    const { toJSON } = render(
      <ThreeVerticalDots displayDatas={['Delete Review', 'Edit Response']} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
