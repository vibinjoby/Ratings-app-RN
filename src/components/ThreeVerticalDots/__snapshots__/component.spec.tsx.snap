// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreeVerticalDots should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 300,
      "position": "absolute",
      "right": 0,
      "width": 300,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/moreIc/moreIc.png",
        }
      }
    />
  </View>
</View>
`;
